isoimage/maker.sh

```diff
@@ -1,17 +1,25 @@
 #!/bin/bash -v
 # The script takes a standard RaspiOS Lite image, installs SamplerBox on it, and creates a ready-to-use image.
-# Notes: 
+# Notes:
 # * this script works on Pi4 but not on Pi2 - tested 2022-08-09
 # * the process is quite long, 1 hr 40 min on a Pi4 - for this reason, I usually start it from screen (screen -S maker, sudo ./maker.sh, CTRL A D to detach)
 #
 # SamplerBox (https://www.samplerbox.org)
 # License: Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International (CC BY-NC-SA 4.0) (https://creativecommons.org/licenses/by-nc-sa/4.0/)
 
+export RASPIOS_ARCH=armhf
+export RASPIOS_NAME=buster
+export RASPIOS_DATE_IMG=2021-05-07
+export RASPIOS_DATE_DIR=2021-05-28
+
+export RASPIOS_BASENAME="${RASPIOS_DATE_IMG}-raspios-${RASPIOS_NAME}-${RASPIOS_ARCH}-lite"
+export RASPIOS_URL="https://downloads.raspberrypi.org/raspios_lite_${RASPIOS_ARCH}/images/raspios_lite_${RASPIOS_ARCH}-${RASPIOS_DATE_DIR}/${RASPIOS_BASENAME}.zip"
```

Comment:
Thanks @mojca.
Good idea, but adding so many variables adds somehow unnecessary complexity.
I would be ok for 3 variables:

    RASPIOS_URL
    RASPIOS_ZIP
    RASPIOS_IMG

Could you modify like this? Then I'll merge it.

Reply:
Just to clarify it first. It's not about "so many variables". The actual variables are "only" 4:
```bash
export RASPIOS_ARCH=armhf
export RASPIOS_NAME=buster
export RASPIOS_DATE_IMG=2021-05-07
export RASPIOS_DATE_DIR=2021-05-28
```

I consider the rest to be the code that users shouldn't ever be touching:
```bash
export RASPIOS_BASENAME="${RASPIOS_DATE_IMG}-raspios-${RASPIOS_NAME}-${RASPIOS_ARCH}-lite"
export RASPIOS_URL="https://downloads.raspberrypi.org/raspios_lite_${RASPIOS_ARCH}/images/raspios_lite_${RASPIOS_ARCH}-${RASPIOS_DATE_DIR}/${RASPIOS_BASENAME}.zip"
```

My main motivation to refactor this code was for the sake of changing it: I really really really wanted to use `aarch64` and a newer version of the OS (I didn't want to keep working with what is now already 3 years old, and I really wanted the 64-bit architecture, so that I could speedily build the image on macOS).

In your original code I would have to replace just `armhf` with `aarch64` at seven different places, and I would have to replace six dates and six occurrences of `buster`.

I gladly change the code, but I don't yet fully understand how, and anything else that currently comes to my mind will make the changes less intuitive for the user to modify.

If I do try to reduce the number of variables from those four, I could do this:
```bash
export RASPIOS_ARCH=armhf
export RASPIOS_BASENAME="2021-05-07-raspios-buster-${RASPIOS_ARCH}-lite"

export RASPIOS_URL="https://downloads.raspberrypi.org/raspios_lite_${RASPIOS_ARCH}/images/raspios_lite_${RASPIOS_ARCH}-2021-05-28/${RASPIOS_BASENAME}.zip"
```
But that's braindead to figure out what needs modifications. This is pretty useless.

The only other way that I can think of would be:
```bash
export RASPIOS_BASENAME="2021-05-07-raspios-buster-armhf"
export RASPIOS_URL="https://downloads.raspberrypi.org/raspios_armhf/images/raspios_armhf-2021-05-28"
```
This would still require changing `armhf` into `aarch64` three times, but maybe it would be "somewhere in the middle" as far as complexity and clarify and effort are concerned. And once these two get defined, the rest of the code doesn't need to be touched any longer.

Please also note that all newer images come with `.xz` rather than `.zip`. So there are further modifications needed anyway. But the mix of code and URLs that had to be changed in order to update the image were the most painful ones. Replacing `zip` by `xz` wasn't that bad.